.github/workflows/internal-archive-release.yml

```diff
@@ -76,30 +124,47 @@ jobs:
         with:
           ref: ${{ inputs.sha || 'main' }}
 
-      - name: "Install Spark Dependencies"
-        if: ${{ contains(github.repository, 'dbt-labs/dbt-spark') }}
-        run: |
-          sudo apt-get update
-          sudo apt-get install libsasl2-dev
-
       - name: "Set up Python - ${{ env.PYTHON_TARGET_VERSION }}"
         uses: actions/setup-python@v5
         with:
           python-version: ${{ env.PYTHON_TARGET_VERSION }}
 
+      #
+      # Testing with Tox
+      #
       - name: "Install Python Dependencies"
+        if: "${{ needs.initial-setup.outputs.is_hatch_adapter == 'false' }}"
+
         run: |
           python -m pip install --user --upgrade pip
           python -m pip install tox
           python -m pip --version
           python -m tox --version
 
-      - name: "Run Tox"
+      - name: "Run Tests using tox"
+        if: "${{ needs.initial-setup.outputs.is_hatch_adapter == 'false' }}"
         run: tox
 
+      #
+      # Testing with Hatch
+      #
+      - name: "Setup `hatch`"
+        if: "${{ needs.initial-setup.outputs.is_hatch_adapter == 'true' }}"
+        uses: dbt-labs/dbt-adapters/.github/actions/setup-hatch@main
+
+      - name: "Run Tests using hatch"
+        if: "${{ needs.initial-setup.outputs.is_hatch_adapter == 'true' }}"
+        run: hatch run unit-tests:all
+
+  ####################
+  # Integration Tests
+  ####################
+
   run-integration-tests:
+    name: 'Integration Tests'
     runs-on: ubuntu-latest
-    needs: [job-setup]
+    needs: [initial-setup, job-setup]
+    if: "${{ needs.initial-setup.outputs.is_hatch_adapter == 'false' }}"
```

Comment:
Why are unit tests a single job but integration tests two jobs?  You should use the separate job model for both.  It makes it easy to see what runs/what gets skipped at the summary level of the workflows.

Reply:
I figured I'd get the opposite criticism for too many jobs